scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/DefaultIllumina.java

```diff
@@ -221,7 +228,39 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
     int exitcode;
     try (InputStream output = process.getInputStream();
         OutputStream input = process.getOutputStream()) {
-      dto = createObjectMapper().readValue(output, IlluminaNotificationDto.class);
+      SimpleModule module = new SimpleModule("customInstantParsingModule");
+
+      module.addSerializer(
+          Instant.class,
+          new JsonSerializer<Instant>() {
+            @Override
+            public void serialize(
+                Instant instant, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+                throws IOException {
+              jsonGenerator.writeString(instant.atZone(tz.toZoneId()).toString());
+            }
+          });
+
+      module.addDeserializer(
+          Instant.class,
+          new JsonDeserializer<Instant>() {
+            @Override
+            public Instant deserialize(
+                JsonParser jsonParser, DeserializationContext deserializationContext)
+                throws IOException, JsonProcessingException {
+              String inststr = jsonParser.getText();
+              try {
+                return ZonedDateTime.parse(inststr).toInstant();
+              } catch (DateTimeParseException dtpe) {
+                throw new IOException(dtpe);
+              }
+            }
+          });
+
+      dto =
+          new ObjectMapper()
+              .registerModule(module)
```

Comment:
Did you look at the `JavaTimeModule` that might work instead? https://github.com/FasterXML/jackson-modules-java8

If not, could this module be defined in its own class rather than recreating it for every run?

It would also be ideal to have a single `ObjectMapper` configured via Spring and used throughout the application, so it behaves the same everywhere.

Reply:
All the other RunProcessors register an ObjectMapper with the JavaTimeModule. However with the Instant parsing that comes it, it requires that datetime strings be in UTC time, which is not possible to get from the C++ runscanner-illumina output. This custom de/serializer allows for the use of any timezone.

Reply:
Ah, makes sense. Maybe just separate creating the module into a separate method as `process` is already really long

Reply:
Done